### DISCUSSION TITLE
Integration of Gas Meter

### DISCUSSION THREAD

**Discussion by ejg3855** [David00/rpi-power-monitor#152] [Ideas]
David,

I think I brought his up 2 years ago but am pondering building the hardware to do it. I am not sure how to get a device like this [https://jomjol.github.io/AI-on-the-edge-device-docs/](url) to integrate with your grafana setup. I'm basically thinking to take a single digital output on the half-foot reading of a gas meter and do some simple math. Im envisioning to export a digital signal every time that rotates and using the dB of your device to record over time and just put it in a graph like the existing power monitor does.

Thoughts?

**Comment by David00** [comment]
Hey @ejg3855, that is pretty cool! I had no idea you could run something like this on an ESP32.

You can certainly use the same local InfluxDB on the Pi that my power monitor project uses to receive the data produced by that project. You can also use the same Grafana application on the Pi to visualize the data.

I'd be happy to point you to where or how the power monitor gets its data into Influx - just let me know!
